implementations/contracts/ERC725XCore.sol

```diff
@@ -87,14 +73,18 @@ abstract contract ERC725XCore is OwnableUnset, ERC165, IERC725X {
 
     /**
      * @dev check the `operationType` provided and perform the associated low-level opcode.
-     * see `IERC725X.execute(...)`.
+     * see `IERC725X.execute(uint256,address,uint256,bytes)`.
      */
     function _execute(
         uint256 operationType,
         address target,
         uint256 value,
         bytes memory data
     ) internal virtual returns (bytes memory) {
+        if (address(this).balance < value) {
```

Comment:
I suggest to move this line to `_executeCall(target, value, data)` and `deployCreate(..)` because:
- CREATE2 already do the check internally and revert with `"Create2: insufficient balance"` if it's the case.
- STATICCALL and DELEGATECALL have  `MsgValueDisallowedIn.. ` Error.

So let's avoid non-necessary checks and move it to the relevant place

Reply:
amazing suggestion ❤️ 👌